src/main/java/com/spotify/reaper/storage/CassandraStorage.java

```diff
@@ -316,17 +316,20 @@ public boolean updateRepairRun(RepairRun repairRun) {
 
   @Override
   public Collection<RepairRun> getRepairRunsWithState(RunState runState) {
-    // There shouldn't be many repair runs, so we'll brute force this one
-    // We'll switch to 2i if performance sucks IRL
-    Collection<RepairRun> repairRuns = Lists.<RepairRun>newArrayList();
-    ResultSet repairRunResults = session.execute(SELECT_REPAIR_RUN);
-    for(Row repairRun:repairRunResults){
-      if(RunState.valueOf(repairRun.getString("state")).equals(runState)){
-        repairRuns.add(buildRepairRunFromRow(repairRun, repairRun.getUUID("id")));
-      }
-    }
-
-    return repairRuns;
+    return getClusters().stream()
+            // Grab all ids for the given cluster name
+            .map((cluster) -> getRepairRunIdsForCluster(cluster.getName()))
+            // Grab repair runs asynchronously for all the ids returned by the index table
+            .flatMap((Collection<UUID> repairRunIds)
+                    -> repairRunIds.stream()
+                            .map((repairRunId) -> session.executeAsync(getRepairRunPrepStmt.bind(repairRunId))))
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the parentheses around the "repairRunId" parameter [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1611)